src/renderer/store.ts

```diff
@@ -1,19 +1,284 @@
-import { createStore, combineReducers } from 'redux';
-import * as LocalRenderer from '@getflywheel/local/renderer';
-import { IPC_EVENTS } from '../constants';
-import { Preferences } from '../types';
-import { preferencesReducer } from './reducers'
-
-
-const makeStore = async () => {
-	const preferences: Preferences = await LocalRenderer.ipcAsync(
-		IPC_EVENTS.READ_PREFERENCES_FROM_DISK
-	);
-
-	return createStore(
-		combineReducers({ preferences: preferencesReducer }),
-		{ preferences },
-	);
+import { useSelector, TypedUseSelectorHook } from 'react-redux';
+import {
+	configureStore,
+	createSlice,
+	createSelector,
+	PayloadAction,
+} from '@reduxjs/toolkit';
+
+import {
+	Preferences,
+	CachedImageDataBySiteID,
+	SiteImageData,
+	OptimizerStatus,
+	ImageData,
+	FileStatus,
+} from '../types';
+
+interface SiteActionPayload {
+	siteID: string;
+	siteImageData: SiteImageData;
+	error: Error;
+}
+
+function mergeSiteState(state, payload: Partial<SiteActionPayload>, newState?: Partial<SiteImageData>) {
+	const { siteID, siteImageData } = payload;
+
+	state[siteID] = {
+		...state[siteID],
+		...siteImageData,
+		...newState,
+	};
+
+	return state;
+}
+
+function generateInitialSiteState(): Partial<SiteImageData> {
+	return {
+		scanInProgress: false,
+		selectAllFilesValue: true,
+		optimizationStatus: OptimizerStatus.BEFORE,
+		compressionListCompletionPercentage: 0,
+	};
+}
+
+const preferencesSlice = createSlice({
+	name: 'preferences',
+	initialState: {} as Preferences,
+	reducers: {
+		hydratePreferences: (_, action: PayloadAction<Preferences>) => {
+			return action.payload;
+		},
+		stripMetaData: (state, action: PayloadAction<boolean>) => {
+			state.stripMetaData = action.payload;
+			return state;
+		},
+	},
+});
+
+const sitesSlice = createSlice({
```

Comment:
I would've liked to see this split up into separate reducers/actions. This being named `sitesSlice` and putting everything within one file feels convenient now, but it will be a refactor down the road.

Reply:
I agree with this. It's one of those things that's never black and white and is something that has to be constantly accessed as things scale, but there is a lot going on here and I think splitting some of that up will help a lot and avoid a more painful refactor later down the road, like Botz mentioned.

Reply:
Devil's advocate: this slice is centered around the data type `CachedImageDataBySiteID` which is a hash map of image optimizer data by site id. If I understand correctly, `createSlice` is intended to create all the stuff you might need around a "slice" of state and I don't think there is a nice way to slice up `CachedImageDataBySiteID` any further. 🤔 I agree that it's pretty gnarly, but maybe a middle ground solution would be to break up slices/selectors into their own files?

Reply:
Ah yes, valid point. It does make me wonder if splitting out store.ts into 3 files might help organize it a bit. Something like:
1. store.ts (just basic store config and types)
2. preferences.ts
3. sites.ts